drivers/bluetooth/hci/userchan.c

```diff
@@ -93,6 +95,68 @@ static struct net_buf *get_rx(const uint8_t *buf)
 	return NULL;
 }
 
+/**
+ * @brief Decode the length of an HCI H4 packet
+ * @details Decodes packet length according to Bluetooth spec v5.4 Vol 4 Part E
+ * @param buf	Pointer to a HCI packet buffer
+ * @return Length of the HCI packet in bytes, zero if no valid packet found.
+ */
+static uint16_t packet_len(const uint8_t *buf)
+{
+	uint16_t payload_len = 0;
+	uint8_t header_len = 0;
+	const uint8_t type = buf[0];
+	const uint8_t *hdr = &buf[sizeof(type)];
+
+	switch (type) {
+	case H4_CMD: {
+		const struct bt_hci_cmd_hdr *cmd = (const struct bt_hci_cmd_hdr *)hdr;
+
+		/* Parameter Total Length */
+		payload_len = cmd->param_len;
+		header_len = BT_HCI_CMD_HDR_SIZE;
+		break;
+	}
+	case H4_ACL: {
+		const struct bt_hci_acl_hdr *acl = (const struct bt_hci_acl_hdr *)hdr;
+
+		/* Data Total Length */
+		payload_len = sys_le16_to_cpu(acl->len);
+		header_len = BT_HCI_ACL_HDR_SIZE;
+		break;
+	}
+	case H4_SCO: {
+		const struct bt_hci_sco_hdr *sco = (const struct bt_hci_sco_hdr *)hdr;
+
+		/* Data_Total_Length */
+		payload_len = sco->len;
+		header_len = BT_HCI_SCO_HDR_SIZE;
+		break;
+	}
+	case H4_EVT: {
+		const struct bt_hci_evt_hdr *evt = (const struct bt_hci_evt_hdr *)hdr;
+
+		/* Parameter Total Length */
+		payload_len = evt->len;
+		header_len = BT_HCI_EVT_HDR_SIZE;
+		break;
+	}
+	case H4_ISO: {
+		const struct bt_hci_iso_hdr *iso = (const struct bt_hci_iso_hdr *)hdr;
+
+		/* ISO_Data_Load_Length parameter */
+		payload_len =  bt_iso_hdr_len(sys_le16_to_cpu(iso->len));
+		header_len = BT_HCI_ISO_HDR_SIZE;
+	}
```

Comment:
```suggestion
		payload_len =  bt_iso_hdr_len(sys_le16_to_cpu(iso->len));
		header_len = BT_HCI_ISO_HDR_SIZE;
		break;
	}
```

Reply:
Thanks for noticing this, have been changing different parts that I must have deleted the break while refactoring.

Reply:
Yeah, it existed earlier :D